pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,9 +1693,58 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Marshal the manifest into an unstructured object
+				dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(step.Resource.Manifest), 10)
```

Comment:
nit: we do this enough that a `StepCodec` would probably be a welcome abstraction.